lib/components/boot-warning/index.jsx

```diff
@@ -1,9 +1,14 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 
 import './style';
 
 const BootWarning = ({ children }) => (
   <h3 className="boot-warning__message">{children}</h3>
 );
 
+BootWarning.propTypes = {
+  children: PropTypes.node.isRequired,
+};
+
```

Comment:
Do you feel the same way about `PropTypes` that the linter does?

In my experience with Calypso they had a tendency to make things just as bad as they helped in other places: they get out of sync (and the linter don't care) and require the extra maintenance.

Moving to Typescript helped out with prop types because we got some support from the compiler, but I think we made some efforts over the years to remove `propTypes`

Reply:
I'm open to changing the rule. I am also open to Typescript.

Reply:
And by open, I think we should add the ability to use Typescript. I know it isn't quite as powerful when bolted on but the code base isn't that large.

Reply:
it's workable easily at our size. I've explored replacing Babel with TypeScript too, which could be a nice move.